notte/common/credentials/models.py

```diff
@@ -0,0 +1,31 @@
+from dataclasses import dataclass
+from typing import Optional, Protocol
+
+from notte.actions.base import ActionParameterValue
+
+
+@dataclass
+class Credentials:
+    url: str
+    username: str
+    password: str
+
+
+class VaultInterface(Protocol):
```

Comment:
why protocol instead of ABC abstract class ?

Reply:
In the package we use the convention `base.py` for bases classes in a directory (with Base in the class name as well, here it can be BaseVault`. This should be renamed